client/signup/steps/import-url/index.jsx

```diff
@@ -2,171 +2,289 @@
+		this.props.fetchIsSiteImportable( this.props.urlInputValue );
 	};
 
-	goToNextStep = () => {
-		const { siteDetails, urlInputValue } = this.props;
-		SignupActions.submitSignupStep( { stepName: this.props.stepName }, [], {
-			importUrl: urlInputValue,
-			importSiteDetails: siteDetails,
-			themeSlugWithRepo: 'pub/radcliffe-2',
+	validateUrl = () => {
+		const validationMessage = validateImportUrl( this.props.urlInputValue );
+		const isValid = ! validationMessage;
+
+		this.setState( {
+			urlValidationMessage: isValid ? '' : validationMessage,
+			showUrlMessage: true,
 		} );
 
-		this.props.goToNextStep();
+		return isValid;
 	};
 
-	handleInputChange = value => {
-		const validationMessage = isValidUrl( value )
-			? ''
-			: this.props.translate( 'Please enter a valid URL.' );
+	getUrlMessage = () => {
+		if ( this.state.urlValidationMessage ) {
+			return this.state.urlValidationMessage;
+		}
 
-		this.setState( {
-			showValidation: false,
-		} );
+		if ( this.props.isSiteImportableError ) {
+			return this.getIsSiteImportableError();
+		}
 
-		this.props.setNuxUrlInputValue( value );
-		this.props.setValidationMessage( validationMessage );
+		return '';
 	};
 
-	handleInputBlur = () => {
-		this.setState( {
-			showValidation: true,
-		} );
+	getIsSiteImportableError = () => {
+		if ( ! this.props.isSiteImportableError ) {
+			return null;
+		}
+
+		const { isSiteImportableError, translate } = this.props;
+
+		switch ( isSiteImportableError.code ) {
+			case SITE_IMPORTER_ERR_INVALID_URL:
+				return translate( 'This does not appear to be a valid URL or website address.' );
+			// @TODO differentiate an unreachable site from one that's not compatible.
+			case SITE_IMPORTER_ERR_BAD_REMOTE:
+				return translate(
+					"We're not able to reach that site at this time, or it's not compatible with our URL importer."
+				);
+		}
+
+		return translate( 'There was an error with the importer, please try again.' );
 	};
 
-	checkValidation( value ) {
-		const { translate } = this.props;
-		const message = isValidUrl( value )
-			? ''
-			: translate( 'Please enter the full URL of your site.' );
+	exitFlow = target => {
+		this.props.recordTracksEvent( 'calypso_signup_flow_exit', {
+			flow: this.props.flowName,
+			step: this.props.stepName,
+			target,
+		} );
+		page.show( target );
+	};
 
-		this.props.setValidationMessage( message );
-	}
+	recordSupportClicked = helpPage => {
+		this.props.recordTracksEvent( 'calypso_signup_support_clicked', {
+			flow: this.props.flowName,
+			step: this.props.stepName,
+			support_page: helpPage,
+		} );
+	};
 
 	renderContent = () => {
-		const { isInputDisabled, urlInputValidationMessage, urlInputValue, translate } = this.props;
-		const { showValidation } = this.state;
+		const { isLoading, urlInputValue, translate } = this.props;
+		const { showUrlMessage } = this.state;
+		const urlMessage = this.getUrlMessage();
+		const helpPage = 'https://en.support.wordpress.com/import/';
```

Comment:
These constants could be module variables to avoid assigning on each render, though their proximity to their usage here is a good enough reason to keep them here too, what d'ya think?